packages/flutter_test/lib/src/_leak_tracking.dart

```diff
@@ -0,0 +1,67 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:leak_tracker/leak_tracker.dart';
+
+import 'leak_tracking.dart';
+import 'test_async_utils.dart';
+import 'widget_tester.dart';
+
+/// Unit test friendly wrapper for [withLeakTracking] with Flutter specific functionality.
+///
+/// The method will fail with test matcher error if wrapped code contains memory leaks.
+///
+/// See details in documentation for `withLeakTracking` at
+/// https://github.com/dart-lang/leak_tracker/blob/main/lib/src/orchestration.dart#withLeakTracking
+///
+/// The Flutter related enhancements are:
+/// 1. Listens to [MemoryAllocations] events.
+/// 2. Uses `asyncCodeRunner` for async call for leak detection.
+Future<void> withFlutterLeakTracking(
```

Comment:
Isn't this method duplicated from elsewhere in the framework now? Can we remove it from there.

Reply:
Yes, this file will be removed in next PR. Added TODO to that method under flutter/test/foundation.

Reply:
Can we just fix this up in this PR?

Reply:
I would love to do it, but I think it is not possible, because the usage of the new method is in different package. So, it needs the method to be checked in, in order to use it.